src/libs/actions/SignInRedirect.js

```diff
@@ -41,7 +46,10 @@ function clearStorageAndRedirect(errorMessage) {
             if (activeClients && activeClients.length > 0) {
                 Onyx.set(ONYXKEYS.ACTIVE_CLIENTS, activeClients);
             }
-            if (isOffline) {
+
+            // After signing out, set ourselves as offline if we were offline before logging out and we are not forcing it.
+            // If we are forcing offline, ignore it while signed out, otherwise it would require a refresh because there's no way to toggle the switch to go back online while signed out.
+            if (isOffline && !shouldForceOffline) {
```

Comment:
This also seems like a good opportunity to just use the "real" value of the network and just let the sign in screen be not testable. If we were logged in before and had `shouldForceOffline` set to true then we'd be stuck in offline mode and couldn't log in to switch it off?

If we want to be able to use "forced offline mode" on the sign in screen then we can do a more global switch that can be used from anywhere. The way we've done it here we can only set it on or off if we are logged in and looking at the `/settings/preferences` route. I think there's a chance QA will find this confusing and lock themselves out of the app 😄

Reply:
The exact purpose of this change is to prevent the user from being unable to sign in if they log out when shouldForceOffline is true.